src/caa-cases/components/caa-filter/caa-filter.component.ts

```diff
@@ -91,10 +90,30 @@ export class CaaFilterComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   public ngOnChanges(changes: SimpleChanges): void {
-    if (changes.selectedOrganisationUsers && changes.selectedOrganisationUsers.currentValue) {
+    if (changes.selectedOrganisationUsers &&
+        changes.selectedOrganisationUsers.currentValue &&
+        changes.selectedOrganisationUsers.currentValue.length > 0) {
       this.filterSelectedOrganisationUsers().subscribe(filteredAndGroupedUsers => {
         this.filteredAndGroupedUsers = filteredAndGroupedUsers;
       });
+      this.initialiseFilterValuesFromSessionState();
+    }
+  }
+
+  public initialiseFilterValuesFromSessionState(): void {
+    if (this.sessionStateValue) {
+      // Set the case reference number input box value if present in session state
+      const caseReferenceNumber = this.sessionStateValue.caseReferenceNumber && this.sessionStateValue.caseReferenceNumber;
+      this.caaFormGroup.get(this.caseRefFormControl).setValue(caseReferenceNumber);
+      // Set the assignee name input box value if present in session state
+      if (this.caaCasesPageType === CaaCasesPageType.AssignedCases) {
+        const assigneeName = this.sessionStateValue.assigneeName && this.sessionStateValue.assigneeName;
```

Comment:
@johnbenjamin-hmcts - duplicate check for `this.sessionStateValue.assigneeName`

Reply:
@Daniel-Lam - I used this statement to get the `assignee` only if it is defined.